lib/nodejs/lobby.js

```diff
@@ -44,6 +44,24 @@ app.get( "/", function( req, res ) {
   } );
 } );
 
+app.get( "/map-list", function( req, res ) {
+  var mapDir = "documents/ITDG/maps";
+
+  // Ensure that the map directory exists, then get the list of files in the map directory
+  mkdirp( mapDir ).then( function( err ) {
+    if ( err ) {
+      console.error( err );
+      res.status( 500 ).send( err );
+      return;
+    }
+    fs.readdirAsync( mapDir ).then( function( files ) {
+      res.send( {
+        mapFilenames: files || []
+      } );
+    } );
+  } );
+} );
```

Comment:
Here's an example with a `catch` handler at the end and forwarding errors to `done`:

    app.get( "/map-list", function( req, res, next ) {
      var mapDir =
        "documents/ITDG/maps";
      mkdirp( mapDir ).
        then( made => fs.readdirAsync( mapDir ) ).
        then( files => res.send( { mapFilenames: files || [] } ) ).
        catch( err => next( err ) );
    } );